staging/src/k8s.io/code-generator/cmd/informer-gen/generators/factory.go

```diff
@@ -237,10 +250,22 @@ func (f *sharedInformerFactory) InformerFor(obj {{.runtimeObject|raw}}, newFunc
 var sharedInformerFactoryInterface = `
 // SharedInformerFactory provides shared informers for resources in all known
 // API group versions.
+//
+// It is typically used like this:
+// 1. create a factory
+// 2. create one or more informers
+// 3. call Start (from the underlying interface) with a stop channel to create
+//    goroutines which handle the informers
+// 4. close the stop channel
+// 5. optional: WaitForShutdown to ensure that goroutines have stopped
```

Comment:
The advantage of embedding this in the generated code is that all factories will have it in their godoc description.

If I were to add an example, where would it go? Should it replace the doc comment?

Reply:
Right I would add a small example of the correct usage in the doc code block comment

Reply:
Got it. I thought initially that you meant an `ExampleSharedInformerFactory` function in a separate _test.go file. That would have been more complicated.

As I am now touching the documentation, let me make some other enhancements:
- I am moving the comments for the functions from the implementation to the interface, because then they appear in the godoc page.
- I am adding methods that are "hidden" in the internal interface to the user-visible SharedInformerFactory, in particular `Start`. That way users see everything they need to know in one godoc page.